javascript/ql/test/library-tests/TypeTracking/tst.js

```diff
@@ -75,7 +75,7 @@ function useConnection() {
   });
 }
 
-export const exportedConnection = getConnection();
+export const exportedConnection = require("util").deprecate(getConnection(), "don't use this");
```

Comment:
Hm, this test is a little misleading as the argument to `deprecate` should be a function - it just seems to return an empty object otherwise. And we currently don't type-track plain functions (my latest attempt [didn't scale on gecko](https://github.com/github/codeql/pull/3639)), so I'm not sure this change would have the intended effect.

So this probably wasn't the perfect use-case for  `PreCallGraphStep` like I thought it was, but I still feel like it _should_ be. I'm happy to merge like this if it works for flow-summaries, and I'll try and poke at the call graph builder again to make it work.

Could you add a test that actually passes a function to `util.deprecate`? (to show the FN)

Reply:
Ah, that was just me being lazy and trying to shoe-horn this onto an existing test instead of making a new one. Will fix.

It's a good point about `util.deprecate` discarding non-functions, though. That makes it a little suspect as a general-purpose data-flow step, though I can't immediately think of a scenario where it would cause false positives.

Reply:
Tests added; please take a look.